app/src/main/java/com/futurice/freesound/utils/TimeScheduler.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 Futurice GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.futurice.freesound.utils;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import io.reactivex.Scheduler;
+import io.reactivex.functions.BiFunction;
+
+import static com.futurice.freesound.functional.Functions.apply;
+import static io.reactivex.schedulers.Schedulers.computation;
+
+/**
+ * Scheduler used for time based operations.
+ */
+public final class TimeScheduler {
+
+    @Nullable
+    private static volatile BiFunction<Scheduler, String, Scheduler> onTimeHandler = null;
+
+    /**
+     * Returns scheduler for time operations.
+     *
+     * @param tag Tag to be used it tests to mock the scheduler
+     */
+    public static Scheduler time(@NonNull final String tag) {
+        BiFunction<Scheduler, String, Scheduler> f = onTimeHandler;
+        if (f == null) {
+            return computation();
+        }
+        return apply(f, computation(), tag);
+    }
+
+    /**
+     * Sets handler for time scheduler.
+     *
+     * @param handler Handler to be used
+     */
+    public static void setTimeSchedulerHandler(BiFunction<Scheduler, String, Scheduler> handler) {
```

Comment:
Unless you want to enforce non-null and direct people to use `reset` if they use null.